thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+                            if (len(lines[i+1].split(':')[0][1:].strip())==0):
+                                while(len(lines[i][:lines[i].find(':')].strip())>2):
+                                    if((i+1) <= len(lines)):
+                                        i+=1
+                                    else:
+                                        i=i-1
+                                        break
+                                    temp = temp + lines[i][lines[i].find(':'):].strip()[:-1].strip()[:-1].strip()
+                        eselParts[searchTerms[term]] = temp
+        os.remove('/tmp/esel.bin')
+    else:
+        print("errl file cannot be found")
+    
+    return eselParts                
+
+"""
+     sorts the sels by timestamp, then log entry number
+       
+     @param events: Dictionary containing events
+     @return: list containing a list of the ordered log entries, and dictionary of keys
+""" 
+def sortSELs(events):
+    logNumList = []
+    timestampList = []
+    eventKeyDict = {} 
+    eventsWithTimestamp = {}
+    logNum2events = {}
+    for key in events:
+        if key == 'numAlerts': continue
+        if 'callout' in key: continue
+        timestamp = (events[key]['timestamp'])
+        if timestamp not in timestampList:
+            eventsWithTimestamp[timestamp] = [events[key]['logNum']]
+        else:
+            eventsWithTimestamp[timestamp].append(events[key]['logNum'])
+        #map logNumbers to the event dictionary keys
+        eventKeyDict[str(events[key]['logNum'])] = key
+        
+    timestampList = list(eventsWithTimestamp.keys())
+    timestampList.sort()
+    for ts in timestampList:
+        if len(eventsWithTimestamp[ts]) > 1:
+            tmplist = eventsWithTimestamp[ts]
+            tmplist.sort()
+            logNumList = logNumList + tmplist
+        else:
+            logNumList = logNumList + eventsWithTimestamp[ts]
+    
+    return [logNumList, eventKeyDict]
+
+"""
+     parses alerts in the IBM CER format, using an IBM policy Table
+       
+     @param policyTable: dictionary, the policy table entries
+     @param selEntries: dictionary, the alerts retrieved from the bmc
+     @return: A dictionary of the parsed entries, in chronological order
+""" 
+def parseAlerts(policyTable, selEntries, args):
+    eventDict = {}
+    eventNum =""
+    count = 0
+    esel = ""
+    eselParts = {}
+    i2cdevice= ""
+    
+    'prepare and sort the event entries'
+    for key in selEntries:
+        if 'callout' not in key:
+            selEntries[key]['logNum'] = key.split('/')[-1]
+            selEntries[key]['timestamp'] = selEntries[key]['Timestamp']
+    sortedEntries = sortSELs(selEntries)
+    logNumList = sortedEntries[0]
+    eventKeyDict = sortedEntries[1]
+    
+    for logNum in logNumList:
+        key = eventKeyDict[logNum]
+#     for key in selEntries:
+        hasEsel=False
+        i2creadFail = False
+        if 'callout' in key:
+            continue
+        else:
```

Comment:
Don't need the else case: continue in the if body will avoid execution of the else body